daemon/response.go

```diff
@@ -246,6 +246,26 @@ func makeErrorResponder(status int) errorResponder {
 	}
 }
 
+// A FileStream ServeHTTP method streams the snap
+type FileStream struct {
+	FileName string
+	Info     snap.DownloadInfo
+	stream   io.ReadCloser
+}
+
+// ServeHTTP from the Response interface
+func (s FileStream) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/octet-stream")
+	size := fmt.Sprintf("%d", s.Info.Size)
+	w.Header().Set("Content-Length", size)
+	defer s.stream.Close()
+
+	if _, err := io.Copy(w, s.stream); err != nil {
```

Comment:
you should be checking how many bytes are copied here I think (and failing with EOF/[502] if it's too short)

[502]: https://httpstatusdogs.com/502-bad-gateway

Reply:
Added the check, but the problem with reporting errors, in this case, is this: after something is written to ResponseWriter, status is automatically set to 200. Setting w.WriteHeader after this make no effect. Don't know how to report an error in this case propperly

Reply:
good point, i guess all we can do is shuffle things on and let the client figure things out from the content-length (and the hash).

Reply:
yes, this is the way to check it, I think